### PR TITLE
Allow API Name change when not on premise

### PR DESCRIPTION
**Pull request by tharikaGitHub**
This PR allows changing the API Name when not in on premise scenario.

Fixes https://github.com/wso2-enterprise/choreo/issues/5257 and https://github.com/wso2-enterprise/choreo/issues/5259

Integration test added to product-apim https://github.com/wso2/product-apim/pull/11395/files

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java [hunk 1]
```diff
@@ -1626,13 +1626,22 @@
...
-                    "   API_UUID = ? " +
+            "   API_TYPE = ? " +
             " WHERE " +
-            "   API_PROVIDER = ? " +
-            "   AND API_NAME = ? " +
-            "   AND" + " API_VERSION = ? " +
-            "   AND ORGANIZATION = ? ";
+            "   API_UUID = ? ";
+
+    public static final String UPDATE_API_SQL_CHOREO =
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java:1633]: lets remove choreo from here

Comment (tharikaGitHub) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/10605/commits/7cff65a402717e2e7d898e4d71caf383062ea334

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java [hunk 2]
```diff
@@ -692,7 +694,11 @@ public static void validateScopes(API api) throws APIManagementException {
     public static API addAPIWithGeneratedSwaggerDefinition(APIDTO apiDto, String oasVersion, String username,
                                                            String organization)
             throws APIManagementException, CryptoException {
-
+        if (APIUtil.isOnPremResolver()) {
+            String name = apiDto.getName();
+            //replace all white spaces in the API Name
+            apiDto.setName(name.replaceAll("\\s+", ""));
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java:700]: This code will remove name with spaces as well right?

Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java:700]: sorry my mistake. this is for onprem

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**